WalletWasabi/BitcoinCore/Processes/BitcoindRpcProcessBridge.cs

```diff
@@ -138,18 +139,27 @@ public async Task StopAsync(bool onlyOwned)
 
 				try
 				{
+					bool isKilled = false;
+
 					try
 					{
+						// Stop Bitcoin daemon using RPC "stop" command.
+						// The command actually only initiates the bitcoind graceful shutdown procedure.
+						// Our time budget for the bitcoind to stop is given by "ReasonableCoreShutdownTimeout".
 						await RpcClient.StopAsync().ConfigureAwait(false);
 					}
 					catch (Exception ex)
 					{
 						Logger.LogWarning(ex);
 						process.Kill();
+						isKilled = true;
 					}
 
-					Logger.LogDebug($"Wait until the process is stopped.");
-					await process.WaitForExitAsync(cts.Token).ConfigureAwait(false);
+					if (!isKilled)
+					{
+						Logger.LogDebug($"Wait until the process is stopped.");
+						await process.WaitForExitAsync(cts.Token).ConfigureAwait(false);
```

Comment:
This will wait endlessly if the process was killed? If it is killed there is no need to wait, is it enough just to believe?

Reply:
No, `WaitForExitAsync` returns immediately. 

`Process.Kill` does not return any result of the operation so one kind of need to believe it. Or check it oneself. 

> is it enough just to believe?

Well, it would be nice to verify it but then again, what would you do if you kill process and you find out that it was not killed. Handling processes is notoriously source of issues because processes are complex too.